File: Ejercicios Propuestos 1/Ejercicio 4/EJERCICIO_4.cpp
#include <iostream>
#include <cmath>

using namespace std;

int main() {
    int num, temp, digito, cantidad;
    int suma_digitos, promedio, primer_digito, divisor;

    cout << "Ingrese un número" << endl;
    cin >> num;

    temp = num;
    cantidad = 0;
    suma_digitos = 0;

    while (temp > 0) {
        digito = temp % 10;
        suma_digitos += digito;
        cantidad += 1;
        temp = temp / 10;
    }

    promedio = suma_digitos / cantidad;
    cout << "Cantidad de dígitos: " << cantidad << endl;
    cout << "Promedio de dígitos (cantidad de rotaciones): " << promedio << endl;

    for (int i = 1; i <= promedio; i++) {
        divisor = pow(10, cantidad - 1);
        primer_digito = num / divisor;
        num = (num % divisor) * 10 + primer_digito;

        cout << " Rotación " << i << ": " << num << endl;
    }

    return 0;
}

File: Ejercicios Propuestos 1/Ejercicio 4/Ejercicio 4.java
import java.util.Scanner;

public class Ejercicio4 {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        int num, temp, digito;
        int cantidad = 0;
        int sumaDigitos = 0;
        int promedio;
        int primerDigito;
        int divisor;

        System.out.print("Ingrese un número: ");
        num = sc.nextInt();

        temp = num;

        // Calcular cantidad y suma de dígitos
        while (temp > 0) {
            digito = temp % 10;
            sumaDigitos += digito;
            cantidad++;
            temp /= 10;
        }

        promedio = sumaDigitos / cantidad;

        System.out.println("Cantidad de dígitos: " + cantidad);
        System.out.println("Promedio de dígitos: " + promedio);

        // Realizar las rotaciones
        for (int i = 1; i <= promedio; i++) {
            divisor = (int) Math.pow(10, cantidad - 1);

            primerDigito = num / divisor;
            num = (num % divisor) * 10 + primerDigito;

            System.out.println("Rotación " + i + ": " + num);
        }

        sc.close();
    }
}
